### PR TITLE
[12.0][MIG] sale_isolated_quotation

### PR DESCRIPTION
**Pull request by bealdav**
- [x]  rename `is_order` to `order_sequence` to be more explicit

### CODE REVIEW

**Comment by bealdav** [comment]
Tests are OK but these warnings appears https://travis-ci.org/OCA/sale-workflow/jobs/633318769#L1576-L1577

 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (order_id, name) of sale.order() have the same label: Order Reference.

 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (state2, state) of sale.order() have the same label: Status.

**Comment by bealdav** [comment]
FYI @kittiu I rename open_sale_order() to open_duplicated_sale_order() to avoid future collisions names

Ready for review

**Comment by kittiu** [comment]
> Tests are OK but these warnings appears https://travis-ci.org/OCA/sale-workflow/jobs/633318769#L1576-L1577
> 
> WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (order_id, name) of sale.order() have the same label: Order Reference.
> 
> WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (state2, state) of sale.order() have the same label: Status.

I guess, you just have to make name different, can you? I.e., put the dot.

**Comment by kittiu** [comment]
As I see, there is not much different from the original 11.0, right? is_order and quotation_state still not being removed?

Then, If you change the field name "state2" to "quotation_state", you probably need to add migration script from 12.0 -> 13.0 too (because changes in database column).

So, why not leave it as is?

**Comment by kittiu** [comment]
I tested it by hand, I got some issue here,

image

File: sale_isolated_quotation/views/sale_views.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                  context="{'is_order': True}"/>
+            </field>
+            <xpath expr="/form/sheet/notebook/page/field[@name='order_line']" position="attributes">
+                <attribute name="context">{'is_order': is_order}</attribute>
+            </xpath>
+        </field>
+    </record>
+    <!-- Assign is_order to context -->
+    <record id="sale.action_quotations" model="ir.actions.act_window">
+        <field name="context">{'hide_sale': True, 'is_order': False}</field>
```
Comment (kittiu) [inline sale_isolated_quotation/views/sale_views.xml]: Not sure what "hide_sale" is for?

**Comment by kittiu** [comment]
Very minor one. But would be nicer to have title as following, for readability.
```
[12.0][MIG] sale_isolated_quotation
```

**Comment by kittiu** [comment]
> open_duplicated_sale_order

I am fine with this.

**Comment by bealdav** [comment]
> Then, If you change the field name "state2" to "quotation_state", you probably need to add migration script from 12.0 -> 13.0 too (because changes in database column).

The field is related (that I've not read at first read of the code), then not stored in database, then no migration schema

**Comment by bealdav** [comment]
> I guess, you just have to make name different, can you? I.e., put the dot.

Already done

**Comment by bealdav** [comment]
> I tested it by hand, I got some issue here,

You're right the new quotation action is `sale.action_quotations_with_onboarding` Fixed

File: sale_isolated_quotation/models/sale.py [hunk 2]
```diff
@@ -11,19 +11,19 @@ class SaleOrder(models.Model):
         string='Is Order',
         readonly=True,
         index=True,
-        default=lambda self: self._context.get('is_order', False),
+        default=lambda self: self.env.context.get('is_order'),
```
Comment (rousseldenis) [inline sale_isolated_quotation/models/sale.py]: @bealdav Maybe this should be avoided using the `_default_<field>` value in context behaviour ?

Comment (bealdav) [reply sale_isolated_quotation/models/sale.py]: Now removed as requested

File: sale_isolated_quotation/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+* Kitti U. <kittiu@ecosoft.co.th>
+* Robert J Sullivan <robertjonsullivan@gmail.com>
```
Comment (mourad-ehm) [inline sale_isolated_quotation/readme/CONTRIBUTORS.rst:2]: @bealdav, Add yourself to contributors

**Review by mourad-ehm — APPROVED** [review]
Code review

**Review by kevinkhao — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@bealdav Do you plan to squash or let commits as this?

**Comment by bealdav** [comment]
Hi @rousseldenis I squashed commits, thanks

**Pull request merged**